Dockerfile

```diff
@@ -185,7 +186,11 @@ FROM dist-$TARGETARCH as dist
 # Optimized container images do not cary nested payloads
 FROM --platform=linux/amd64 cpu-builder-amd64 AS container-build-amd64
 WORKDIR /go/src/github.com/ollama/ollama
-COPY . .
+# Download the Go module dependencies
```

Comment:
this is an optimization for image rebuilds only. so when we rebuild it doesnt need to redownload dependencies again

should probably be in separate branch/pr